misc/pstring.cpp

```diff
@@ -71,7 +71,7 @@ std::size_t CleanupStr(char *dest, const char *src, std::size_t destlen) {
   end++;
 
   // Set output size to minimum of trimmed string length and buffer size minus 1
-  out_size = (end - src) < destlen - 1 ? (end - src) : destlen - 1;
+  out_size = static_cast<size_t>(end - src) < destlen - 1 ? (end - src) : destlen - 1;
```

Comment:
Consider providing the exact compiler warning message in the commit log so later onlookers can establish that the chosen solution was indeed the right choice.